gremlin-dotnet/test/Gremlin.Net.IntegrationTest/Gherkin/GherkinTestRunner.cs

```diff
@@ -0,0 +1,381 @@
+                    return StepBlock.Given;
+                case Keywords.When:
+                    return StepBlock.When;
+                case Keywords.Then:
+                    return StepBlock.Then;
+                case Keywords.And:
+                case Keywords.But:
+                    if (currentStep == null)
+                    {
+                        throw new InvalidOperationException("'And' or 'But' is not supported outside a step");
+                    }
+                    return currentStep.Value;
+            }
+            throw new NotSupportedException($"Step with keyword {stepKeyword} not supported");
+        }
+
+        private static StepDefinition GetStepDefinitionInstance(IEnumerable<Type> stepDefinitionTypes, string stepText)
+        {
+            var type = stepDefinitionTypes
+                .FirstOrDefault(t => t.GetMethods().Any(m =>
+                {
+                    var attr = m.GetCustomAttribute<GivenAttribute>();
+                    if (attr == null)
+                    {
+                        return false;
+                    }
+                    return Regex.IsMatch(stepText, attr.Message);
+                }));
+            if (type == null)
+            {
+                throw new InvalidOperationException($"No step definition class matches Given '{stepText}'");
+            }
+            return (StepDefinition) Activator.CreateInstance(type);
+        }
+
+        private ICollection<Type> GetStepDefinitionTypes()
+        {
+            var assembly = GetType().GetTypeInfo().Assembly;
+            var types = assembly.GetTypes()
+                .Where(t => typeof(StepDefinition).IsAssignableFrom(t) && !t.GetTypeInfo().IsAbstract)
+                .ToArray();
+            if (types.Length == 0)
+            {
+                throw new InvalidOperationException($"No step definitions in {assembly.FullName}");
+            }
+            return types;
+        }
+
+        private IEnumerable<Feature> GetFeatures()
+        {
+            var rootPath = GetRootPath();
+            var path = Path.Combine(rootPath, "gremlin-test", "features");
+            var files = Directory.GetFiles(path, "*.feature", SearchOption.AllDirectories);
+            foreach (var gherkinFile in files)
+            {
+                var parser = new Parser();
+                var doc = parser.Parse(gherkinFile);
+                yield return doc.Feature;
+            }
+        }
+
+        private string GetRootPath()
+        {
+            var codeBaseUrl = new Uri(GetType().GetTypeInfo().Assembly.CodeBase);
+            var codeBasePath = Uri.UnescapeDataString(codeBaseUrl.AbsolutePath);
+            DirectoryInfo rootDir = null;
+            for (var dir = Directory.GetParent(Path.GetDirectoryName(codeBasePath));
+                dir.Parent != null;
+                dir = dir.Parent)
+            {
+                if (dir.Name == "gremlin-dotnet" && dir.Parent?.Name == "tinkerpop")
```

Comment:
I would prefer if we could make this more flexible so that other names than _tinkerpop_ for the directory are allowed. Since the directory isn't part of the repo it can have an arbitrary name.
(The test runner failed for me because the directory name _TinkerPop_.)

Reply:
Agree, looking for `tinkerpop` directory is incorrect.
I've changed it to look for `gremlin-dotnet` directory which has a `pom.xml` file. There are different approaches we can use (like using an env var) but each has its own downsides.